tests/components/frigidaire/test_config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Test the frigidaire config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.frigidaire.config_flow import CannotConnect, InvalidAuth
+from homeassistant.components.frigidaire.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_FORM
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.frigidaire.config_flow.ConfigFlow.async_step_user",
+        return_value={
+            "type": RESULT_TYPE_FORM,
+            "title": "Name of the device",
+            "flow_id": "",
+            "handler": "",
+            "step_id": "",
+            "data_schema": "",
+            "errors": {},
+            "description_placeholders": "",
+            "last_step": True,
+            "data": {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        },
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "form"
+    assert result2["title"] == "Name of the device"
+    assert result2["data"] == {
+        "username": "test-username",
+        "password": "test-password",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.frigidaire.config_flow.validate_input",
+        side_effect=InvalidAuth,
+    ):
```

Comment:
Mock out the client that raises the side effect resulting in InvalidAuth instead.